### ISSUE TITLE
Support for GTFS Flex

### ISSUE THREAD

**Issue by fchabouis** [rust-transit/gtfs-structure#135]
I never noticed, but the GTFS Flex extension has the specificity to add a [geojson file](https://github.com/MobilityData/gtfs-flex/blob/master/spec/reference.md#locationsgeojson-file-added) in the GTFS specification.
This is still a proposal, so there is no rush here, but I am curious to know your opinion on how difficult it would be to adapt this crate to handle this new kind of format.
Thanks!

@Tristramg @antoine-de

**Comment by Tristramg** [comment]
I don’t think it would be a big problem. We already use `serde` for the csv, and its json support is very good.

**Comment by lolpro11** [comment]
This would be a small update but make stops.txt optional as well to support GTFS Flex

**Comment by robfitzgerald** [comment]
Hello, I have begun working on an implementation to add this feature to gtfs-structures and wanted to gauge interest. I'm a research software engineer at NRL in Colorado, USA (formerly NREL) where I support mobility science research with our open-source route planner and access model ([RouteE Compass](https://github.com/NatLabRockies/routee-compass), [BAMBAM](https://github.com/NatLabRockies/bambam)). our business case is we are developing a GTFS-Flex access model and we currently use gtfs-structures for our transit access model.

my WIP has the following objectives:
  - add GTFS-Flex data models and collections under a gtfs-flex compiler flag
  - no API changes for users with feature disabled
  - use [this examples page](https://gtfs.org/documentation/schedule/examples/flex/) as the source of truth (didn't find a spec page)
  - add tests using new test fixtures from [Brown County Heartland Express](https://mobilitydatabase.org/feeds?q=Brown%20County%20Heartland%20Express) and [Otter Express](https://mobilitydatabase.org/feeds?q=Otter+Express) feeds

my main critique of my design is that the `StopTime` and `Gtfs` types gets a bit sliced up with `#[cfg(feature)]` macros in a way that affects readability. this feels more complicated with `StopTime` which sits in the middle of other types and methods that may or may not be connected based on the compiler flag. the tradeoff being that existing users can have the same API as they do now with the feature deactivated. i _think_ this is the best way to accomplish this, but, i would worry about how this might explode if additional gtfs extensions took a similar approach.

once i wrap up a first draft i would submit it as a draft PR. noting in the discussion above there was some buy-in from a core contributor (@Tristramg), but, that was 3 years ago, so i first want to confirm that there would be interest in this proposed change.

edit: you can see my WIP changes [here](https://github.com/robfitzgerald/gtfs-structure/compare/702130b..bc1968c)

**Comment by robfitzgerald** [comment]
hey folks, i'm using my fork of your repo for my GTFS-Flex modeling needs, but just wanted to ping this discussion again to see if there was any interest in a PR. thank you for your work.

**Comment by antoine-de** [comment]
Oups sorry, I missed your message. Neither @Tristramg nor me are currently working on this lib, but we'll happily review a merge request if it can help others! 

For the feature gating, I don't really know, it's true that the readibility is impacted, but in the meantime it can be nice to limit the number of fields in StopTime since it will have the main impact on memory consumption, so  I guess you can open a MR and we'll discuss it there?

**Comment by robfitzgerald** [comment]
@antoine-de thanks for getting back! i'm in a busy week this week, and since i no longer remember all of the changes i made, i want to review my own work before i set up the PR, so it may be a week or so until that happens. :bus:
